app/assets/javascripts/global.js

```diff
@@ -13,8 +13,14 @@ $(document).ready(function() {
     return false;
   });
 
-  $("#select_all").click(function() {
-    $(".select-all-box").prop('checked', this.checked);
+  $('.check-all').on('change', function() {
+    // The check-all's checkbox value will contain the name of the checkboxes which are meant to be checked
+    $(`.checkbox[name="${this.value}"]`).prop('checked', this.checked);
+  });
+
+  $('.checkbox').on('change', function() {
+    const allChecked = $(`.checkbox[name="${this.name}"]`).filter(':checked').length === $(`.checkbox[name="${this.name}"]`).length;
```

Comment:
Are all `.checkboxes` part of a series for `.check-all`? It seems like it would be safer to scope this JavaScript only to checkboxes that we know are part of check-all lists. It likely doesn't do anything in the case there isn't a corresponding `.check-all`, but I'm thinking if `this.name` might be empty for some checkboxes or if there are some other weird edge cases here.

Reply:
`this.name` works fine if it's empty and that does not do anything when there isn't a matching `.check-all`

Reply:
(discussed aloud, seems to be true but would be nice to avoid having to think this through to be sure of the correctness)